Configurations/unix-Makefile.tmpl

```diff
@@ -504,23 +504,31 @@ LANG=C
 
 # The main targets ###################################################
 
+##@ Software
+build_sw: ## Build all the software (default target)
+build_libs: ## Build the libraries libssl and libcrypto
+build_modules: ## Build the modules (i.e. providers and engines)
+build_programs: ## Build the openssl executables and scripts
+all: build_sw build_docs ## Build software and documentation
+
 {- dependmagic('build_sw'); -}: build_libs_nodep build_modules_nodep build_programs_nodep link-utils
 {- dependmagic('build_libs'); -}: build_libs_nodep
 {- dependmagic('build_modules'); -}: build_modules_nodep
 {- dependmagic('build_programs'); -}: build_programs_nodep
```

Comment:
Er...  are you sure about this?  The perl snippets with the `dependmagic()` call generate targets as well, the exact same as You've added above.  Doesn't that become a conflict?

You might want to expand the `dependmagic()` sub to take the documentation string as a second argument, and ensure that it becomes part of the generated target...

Reply:
No, it does not create a conflict. It simply adds no additional dependencies. All the builds passed.

Reply:
I know it looks strange, but it isn't that much different from something like:
```
build_sw: TMP_VAR=nothing_useful
```
Which just defines a variable for that build. But in this case, nothing additional is added. If we do have a concern, I can try to make these into variables, which would be ignored during the build.

Reply:
So, not exactly, but it acts as a force target for things that don't support `.PHONY` https://www.gnu.org/software/make/manual/html_node/Force-Targets.html

Reply:
And also, when setting variables:
https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html

Reply:
Let me fix this.